lib/Test2/Compare/Bool.pm

```diff
@@ -0,0 +1,111 @@
+package Test2::Compare::Bool;
+use strict;
+use warnings;
+
+use Carp qw/confess/;
+
+use base 'Test2::Compare::Base';
+
+our $VERSION = '0.000059';
+
+use Test2::Util::HashBase qw/input/;
+
+# Overloads '!' for us.
+use Test2::Compare::Negatable;
+
+sub name {
+    my $self = shift;
+    my $in = $self->{+INPUT};
+    return _render_bool($in);
+}
+
+sub operator {
+    my $self = shift;
+    return '!=' if $self->{+NEGATE};
+    return '==';
+}
+
+sub verify {
+    my $self = shift;
+    my %params = @_;
+    my ($got, $exists) = @params{qw/got exists/};
+
+    return 0 unless $exists;
+
+    my $want = $self->{+INPUT};
+
+    my $match = ($want xor $got) ? 0 : 1;
+    $match = $match ? 0 : 1 if $self->{+NEGATE};
+
+    return $match;
+}
+
+sub run {
+    my $self = shift;
+    my $delta = $self->SUPER::run(@_) or return;
+
+    my $dne = $delta->dne || "";
+    unless ($dne eq 'got') {
+        my $got = $delta->got;
+        $delta->set_got(_render_bool($got));
+    }
+
+    return $delta;
+}
+
+sub _render_bool {
+    my $bool = shift;
+    my $name = $bool ? 'TRUE' : 'FALSE';
+    my $val = defined $bool ? $bool : 'undef';
+    $val = "''" unless length($val);
+
+    return "$name ($val)";
```

Comment:
I'd even like to see something like `true(5)` and `false(undef)`.  At least that's how I've set it up in my original `bool_eq`.  I want to see both the base value and what it evaluates to.